ivadomed/models.py

```diff
@@ -1486,27 +1485,40 @@ def set_model_for_retrain(model_path, retrain_fraction, map_location, reset=True
 def get_model_filenames(folder_model):
     """Get trained model filenames from its folder path.
 
-    This function checks if the folder_model exists and get trained model (.pt or .onnx) and its configuration file
-    (.json) from it.
-    Note: if the model exists as .onnx, then this function returns its onnx path instead of the .pt version.
+    This function checks if the folder_model exists and get trained model path (.pt or .onnx based on
+    model and GPU availability) and its configuration file (.json) from it.
 
     Args:
         folder_name (str): Path of the model folder.
 
     Returns:
-        str, str: Paths of the model (.onnx) and its configuration file (.json).
+        str, str: Paths of the model (.pt or .onnx) and its configuration file (.json).
     """
     if Path(folder_model).is_dir():
+
         prefix_model = Path(folder_model).name
-        # Check if model and model metadata exist. Verify if ONNX model exists, if not try to find .pt model
-        fname_model = Path(folder_model, prefix_model + '.onnx')
-        if not fname_model.is_file():
-            fname_model = Path(folder_model, prefix_model + '.pt')
-            if not fname_model.exists():
-                raise FileNotFoundError(fname_model)
+        fname_model_onnx = Path(folder_model, prefix_model + '.onnx')
+        fname_model_pt = Path(folder_model, prefix_model + '.pt')
+        cuda_available = torch.cuda.is_available()
+
+        # Assign '.pt' or '.onnx' model based on file existence and GPU/CPU device availability
+        if not fname_model_pt.is_file() and not fname_model_onnx.is_file():
+            raise FileNotFoundError(f"Model files not found in model folder: "
+                                    f"'{str(fname_model_onnx)}' or '{str(fname_model_pt)}'")
+        # '.pt' is preferred on GPU, or on CPU if '.onnx' doesn't exist
+        elif ((    cuda_available and     fname_model_pt.is_file()) or
```

Comment:
Not sure if the spacing after `((` and `and` is intended but if not we can remove them. Same thing for lines 1513 and 1514 as well.

Reply:
It's definitely nonstandard formatting, aha, so if you feel like removing it, go for it.

The reason I added it in the first place was because I thought it made it a little easier to parse (since everything lined up nicely, and it was clear whether or not `not` was present.)

Reply:
Ah I see, no worries if it has a purpose :)